lib/workload/stateless/stacks/bssh-icav2-fastq-copy-manager/deploy/stack.ts

```diff
@@ -0,0 +1,84 @@
+import * as cdk from 'aws-cdk-lib';
+import { Construct } from 'constructs';
+import * as ssm from 'aws-cdk-lib/aws-ssm';
+import * as sfn from 'aws-cdk-lib/aws-stepfunctions';
+import { BsshIcav2FastqCopyStateMachineConstruct } from './constructs/bssh-icav2-fastq-copy-manager';
+import * as secretsmanager from 'aws-cdk-lib/aws-secretsmanager';
+import path from 'path';
+import { PythonLambdaLayerConstruct } from '../../../../components/python-lambda-layer';
+
+export interface BsshIcav2FastqCopyManagerConfig {
+  // Define construct properties here
+  icav2_jwt_secrets_manager_path: string; // "/icav2/umccr-prod/service-production-jwt-token-secret-arn"
+  icav2_copy_batch_utility_state_machine_name: string; // "/icav2/umccr-prod/copy-batch-state-machine/name"
+  bssh_icav2_fastq_copy_manager_state_machine_name: string; // "bssh_icav2_fastq_copy_manager"
+  bssh_icav2_fastq_copy_manager_state_machine_name_ssm_parameter_path: string; // "/bssh_icav2_fastq_copy/state_machine/name"
+  bssh_icav2_fastq_copy_manager_state_machine_arn_ssm_parameter_path: string; // "/bssh_icav2_fastq_copy/state_machine/arn"
+}
+
+export type BsshIcav2FastqCopyManagerStackProps = BsshIcav2FastqCopyManagerConfig & cdk.StackProps;
+
+export class BsshIcav2FastqCopyManagerStack extends cdk.Stack {
+  public readonly bssh_icav2_fastq_copy_state_machine_arn: string;
+
+  constructor(scope: Construct, id: string, props: BsshIcav2FastqCopyManagerStackProps) {
+    super(scope, id, props);
+
+    // Get the copy batch state machine value from lookup
+    const icav2_copy_batch_stack_state_machine_arn_obj = sfn.StateMachine.fromStateMachineName(
+      this,
+      'icav2_copy_batch_state_machine',
+      props.icav2_copy_batch_utility_state_machine_name
+    );
```

Comment:
Alternatively. 

We should send `manifest_b64gz` back into `MainBus` channel; let `ICAv2CopyBatchUtility` service pick it up and, do its job.

That'd create an event schema, say, `ICAv2CopyBatchUtilityRequest` for example.

In this case, we just have to watch out on EventBridge event message size and number of characters limit.

- https://www.google.com/search?q=eventbridge+message+size+limit
- https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_Target.html
- https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-putevent-size.html

Reply:
> I think, this makes dependency on ICAv2CopyBatchUtilityStack (deployment aspect) and, service that stack creates (app aspect) - this is okay, expected.

Just using the name as defined in the constants, rather than relying on an SSM parameter path, but yes the lookup based on the state machine name will fail if it doesn't exist. Not sure if this is a bad thing given the dependency.  

> We should send manifest_b64gz back into MainBus channel; let ICAv2CopyBatchUtility service pick it up and, do its job.

Hmm, depends how far down the events rabbit hole we want to go.  This might work in this case because the bssh fastq copy manager's last step is calling the copy batch utility. Completion of the copy batch utility could then raise an internal event that lets bssh fastq copy know it can now raise its own completion event.  

However, we use this same service for the cttsov2 pipeline.  It takes about 3 seconds for the tso fastqs to transfer to their new directory, the analysis is then started after the fastq transfers are complete.  If we went with an event based structure for this, the cttso step function would need to submit the event, pause the workflow and then resume for the analysis submission once it hears back from the copy batch utility service.  I get the desire to go with the event-based architecture, but I don't think it completely decouples things.

In hindsight, the icav2 copy batch utility service could probably have been part of the infrastructure repository and a static service that can be called on, like the credentials rotation stack

Reply:
We learn... 

I can throw another option into the mix: create your own (internal) ICAv2CopyBatchUtility exclusively for this stack's purpose. No dependency on existing/external stacks.

Reply:
😭

Reply:
HeHe, not saying we should refactor. Just keeping in mind that not all duplication is bad... it can have it's advantages.